### ISSUE TITLE
Support external .lrc file

### ISSUE THREAD

**Issue by Extarys** [navidrome/navidrome#1421]
### Is your feature request related to a problem? Please describe.

The PR #1406 is awesome as I love lyrics, but all my synched lyrics are in a file ending with `.lrc` and not inside the metadata of the song.

### Describe the solution you'd like

I suggest that either on library scan, or when a song is started, navidrome look up if there is a file named like the song being played, but ending in .lrc `01 - My song.flac` > `01 - My song.lrc`

### Additional context

On PC I currently use Tauon when I want to see lyrics, sadly my "away" music player (Jellyfin) doesn't support lyrics at all.

**EDIT** : Sorry I just saw this comment a little too late, closing. https://github.com/navidrome/navidrome/issues/491#issuecomment-946782864

**Comment by dheerajdlalwani** [comment]
I am starting to work on this issue now.

**Comment by chermy** [comment]
Hopefully this can be added soon. Thanks for all of your work guys ;-D

**Comment by certuna** [comment]
Maybe worth considering that there's also `.elrc` files, with per-word timestamps: https://stage.elsdoerfer.com/elrcmaker/

**Comment by Extarys** [comment]
@certuna Those are way harder to make though and I never caught a wild one - but yes they esists and it's dope :D

**Comment by KrLx1994roller** [comment]
@dheerajdlalwani how is the implementation of this going? Do you need any help? :)

**Comment by dheerajdlalwani** [comment]
Hey, @KrLx1994roller ! Thank you for asking and I'm so sorry for the late replies. I haven't been actively working on this. It completely slipped off my mind because of college assignments and projects.

**Comment by Extarys** [comment]
@mrsobakin  You mean if the file was modified? I would calculate a CRC32 checksum and compare to the previous file or check last modified date.

Are you working on this? :heart:

**Comment by mrsobakin** [comment]
@Extarys No, I mean detecting the presence of a `.lrc` file. You see, when navidrome performs the scan of a directory, it only checks \*updated\* \*audio\* files. Imagine this situation: You already have some track in navidrome library and you decide to add a `.lrc` file for it. Navidrome performs scan of changed directory, but finds nothing, because the track file didn't change, and the `.lrc` is not an audio. 

There is an obvious solution - scan directory for `.lrc` files and store them in database, but as far as I know, it would require changing the database structure, which:
1. Seems like an overkill for this.
2. I don't feel competent enough doing.

**Comment by Extarys** [comment]
When playing a song, check if the lrc file is there. If it's there, load it in the browser. This will require an additionnal file seek for every song played though, but I guess this is how most players implement this.

Tauon have a row in the GUI that shows if there are lyrics present, that would require a new row in the DB like `hasLrc` set to false by default and updated to true on library scan when one is found

**Comment by mrsobakin** [comment]
Okay, so I think I came up somewhat good solution... We will load external lyrics in `metadata.Extract()`, and we will also change `processChangedDir()` to update track if corresponding lyrics file gets updated. The `updated_at` row will be used for both lyrics and track files. Also, since they share stem (name w/o extension), we don't have to create any new rows.

The more I think about it, the more it seems that we should just go the overkill way and create a table for `.lrc` files. It may seem more complicated, but at least it's not janky.

I'll begin to experiment with the this "janky" solution, but I really think we should discuss the way external lyrics will be implemented.

**Comment by mrsobakin** [comment]
The solution I described above works.. Almost. Because we store both `.lrc` and track update time in a single `updated_at` row, lyrics aren't "unloaded" when the corresponding `.lrc` file is being deleted. Otherwise, it works as expected.

To this, we have 2 solutions:
1. More kludges.
2. Redesign of scanning system and modifying the db.

Both of which is.. meh

**Comment by Extarys** [comment]
I would argue that modifying the DB is the best future-proof option here, but I'm not the one doing the heavy-lifting, so it's still up to you :smile:

**Comment by xinmans** [comment]
hope can support next versioin

**Comment by KITblue** [comment]
现在发现歌词切换经常错误，当第一首歌正常播放完，切换第二首歌时，歌词切换会出错，有时仍然显示上一首歌的歌词，有时候显示的是这一首歌的但时间是错误的，打开播放列表，点击重新播放又能正常显示了，

使用的是docker版本

**Comment by smmya** [comment]
Thank you very much to all the developers for their hard work in developing and perfecting this project.
I wonder when external ".lrc" files will be supported.
I have nearly two thousand songs in my music library, and their lyrics are all in lrc files of the same name, they are all stored in cloud services, it would be a big project to pull it back and repackage the lyrics files, External lrc lyrics file support can make maintaining music library easier, hope this can be resolved.

**Comment by deluan** [comment]
Yes it will be, at some point, but it is not a priority ATM. As always, Pull Requests are mostly welcome :)

**Comment by wq2020wdm** [comment]
Really Looking forward to that function. I just downloaded about 20,000 songs from PT website. All are external ".lrc". I can not change the files because I need keep seeding. By the way, the navidrome is wonderful and meets my requirements. Thanks for your working.

**Comment by eatgrass** [comment]
any news about .lrc support?

**Comment by IceWreck** [comment]
I build a python cli tool, https://github.com/IceWreck/tagpatch to take my lrc files and embed them inside corresponding music file automatically.
`tagpatch embed-lrc -n -s ~/Music`, look at your changes, confirm and you're done.

And now I can view lyrics inside navidrome.

**Comment by zrwd01** [comment]
@deluan great job

**Comment by Batwam** [comment]
I'd love to see this too, especially since LRCGET only seems to save the lyrics in .lrc files

**Comment by TacoCake** [comment]
FIY, related PR: https://github.com/navidrome/navidrome/pull/2897

> - Updates local agent to read lyrics from track.txt/track.lrc

**Comment by Cipheritall** [comment]
I got you covered y all o/
https://github.com/navidrome/navidrome/pull/3629

**Comment by qndydyhm** [comment]
> I got you covered y all o/ #3630

Thanks a lot! Just started trying navidrome and saw this PR!

**Comment by deluan** [comment]
Support for .lrc was added in https://github.com/navidrome/navidrome/pull/2897, will be part of next release.

**Comment by a0g83agbc84** [comment]
that's crazy good news, thanks! a question about its behaviour: in case the audio file already has embedded lyrics within their metadata *and* the `.lrc` file is sitting by the audio file, how will navidrome handle this? using the embedded one or using the external one?

**Comment by Siscos06** [comment]
> Ohh wow. Next release. I hope it would be soon. I almost gave up on navidrome cuz i installed it today (new user) and it wasn't parsing lyrics from lrc. Navidrome is indeed best when it comes to metadata parsing. Didn't wanna lose that. Thanks for the addition

Has it been added? I'm willing to parse my lyrics and I'm looking for options.
